src/pages/ClusterPage/utils/index.js

```diff
@@ -547,3 +547,78 @@ export const hasAddon = (item, source) =>
 	!!(source.addons || []).find(key => key.name === item);
 export const getAddon = (item, source) =>
 	(source.addons || []).find(key => key.name === item);
+const generateRandomString = length => {
+	const chars =
+		'0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ';
+	let result = '';
+	for (let i = length; i > 0; i -= 1)
+		result += chars[Math.floor(Math.random() * chars.length)];
+	return result;
+};
+
+export const rotateAPICredentials = (type, clusterId) => {
+	const username = generateRandomString(9);
+	const password = `${generateRandomString(8)}-${generateRandomString(
```

Comment:
We use a UUID generation util in the BE. That's more failproof than doing this imo.

Reply:
@siddharthlatest I have used uuid v4 to generate the password